File: Intersection of Two Linked Lists.js
var getIntersectionNode = function(headA, headB) {
    if (headA == null || headB == null) return null;
        var listA = headA, listB = headB;
        while (listA != null && listB != null) {
            if (listA == listB) return listA;
            listA = listA.next;
            listB = listB.next;
            if (listA == listB) return listA;
            if (listA == null) listA = headB;
            if (listB == null) listB = headA;
        }
        return listA;
};

File: Intersection of Two Linked Lists.c
struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB) {
    if (headA == NULL || headB == NULL) return NULL;
        struct ListNode * listA = headA, *listB = headB;
        while (listA != NULL && listB != NULL) {
            if (listA == listB) return listA;
            listA = listA->next;
            listB = listB->next;
            if (listA == listB) return listA;
            if (listA == NULL) listA = headB;
            if (listB == NULL) listB = headA;
        }
        return listA;
}
